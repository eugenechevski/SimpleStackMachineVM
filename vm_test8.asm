	# $Id: vm_test8.asm,v 1.4 2024/08/15 18:58:50 leavens Exp $
	.text 0
	LIT $sp, 0, 17
	LWR $r4, $sp, 0   # put 14 in $r4
	SRI $sp, 2
	LIT $sp, 0, 19
	LWR $r5, $sp, 0   # put 19 in $r5
	SCA $sp, -1, $sp, 0 # store stack top address
	LWI $sp, -2, $sp, -1
	SRI $sp, 2
	EXIT 0
	.data 256
	.stack 1024
	.end
