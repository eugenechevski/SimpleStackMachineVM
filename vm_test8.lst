Address Instruction
     0: LIT $sp, 0, 17
     1: LWR $r4, $sp, 0
     2: SRI $sp, 2
     3: LIT $sp, 0, 19
     4: LWR $r5, $sp, 0
     5: SCA $sp, -1, $sp, 0
     6: LWI $sp, -2, $sp, -1
     7: SRI $sp, 2
     8: EXIT 0
     256: 0	        ...     
