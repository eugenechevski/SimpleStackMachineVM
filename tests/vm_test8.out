      PC: 0
GPR[$gp]: 256  	GPR[$sp]: 1024 	GPR[$fp]: 1024 	GPR[$r3]: 0    	GPR[$r4]: 0    
GPR[$r5]: 0    	GPR[$r6]: 0    	GPR[$ra]: 0    
     256: 0	        ...     
    1024: 0	

==>      0: LIT $sp, 0, 17
      PC: 1
GPR[$gp]: 256  	GPR[$sp]: 1024 	GPR[$fp]: 1024 	GPR[$r3]: 0    	GPR[$r4]: 0    
GPR[$r5]: 0    	GPR[$r6]: 0    	GPR[$ra]: 0    
     256: 0	        ...     
    1024: 17	

==>      1: LWR $r4, $sp, 0
      PC: 2
GPR[$gp]: 256  	GPR[$sp]: 1024 	GPR[$fp]: 1024 	GPR[$r3]: 0    	GPR[$r4]: 17   
GPR[$r5]: 0    	GPR[$r6]: 0    	GPR[$ra]: 0    
     256: 0	        ...     
    1024: 17	

==>      2: SRI $sp, 2
      PC: 3
GPR[$gp]: 256  	GPR[$sp]: 1022 	GPR[$fp]: 1024 	GPR[$r3]: 0    	GPR[$r4]: 17   
GPR[$r5]: 0    	GPR[$r6]: 0    	GPR[$ra]: 0    
     256: 0	        ...     
    1022: 0	        ...         1024: 17	

==>      3: LIT $sp, 0, 19
      PC: 4
GPR[$gp]: 256  	GPR[$sp]: 1022 	GPR[$fp]: 1024 	GPR[$r3]: 0    	GPR[$r4]: 17   
GPR[$r5]: 0    	GPR[$r6]: 0    	GPR[$ra]: 0    
     256: 0	        ...     
    1022: 19	    1023: 0	    1024: 17	

==>      4: LWR $r5, $sp, 0
      PC: 5
GPR[$gp]: 256  	GPR[$sp]: 1022 	GPR[$fp]: 1024 	GPR[$r3]: 0    	GPR[$r4]: 17   
GPR[$r5]: 19   	GPR[$r6]: 0    	GPR[$ra]: 0    
     256: 0	        ...     
    1022: 19	    1023: 0	    1024: 17	

==>      5: SCA $sp, -1, $sp, 0
      PC: 6
GPR[$gp]: 256  	GPR[$sp]: 1022 	GPR[$fp]: 1024 	GPR[$r3]: 0    	GPR[$r4]: 17   
GPR[$r5]: 19   	GPR[$r6]: 0    	GPR[$ra]: 0    
     256: 0	        ...         1021: 1022	
    1022: 19	    1023: 0	    1024: 17	

==>      6: LWI $sp, -2, $sp, -1
      PC: 7
GPR[$gp]: 256  	GPR[$sp]: 1022 	GPR[$fp]: 1024 	GPR[$r3]: 0    	GPR[$r4]: 17   
GPR[$r5]: 19   	GPR[$r6]: 0    	GPR[$ra]: 0    
     256: 0	        ...         1020: 19	    1021: 1022	
    1022: 19	    1023: 0	    1024: 17	

==>      7: SRI $sp, 2
      PC: 8
GPR[$gp]: 256  	GPR[$sp]: 1020 	GPR[$fp]: 1024 	GPR[$r3]: 0    	GPR[$r4]: 17   
GPR[$r5]: 19   	GPR[$r6]: 0    	GPR[$ra]: 0    
     256: 0	        ...     
    1020: 19	    1021: 1022	    1022: 19	    1023: 0	    1024: 17	

==>      8: EXIT 0
